### PR TITLE
Update local dev setup

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement
/priority normal

**What this PR does / why we need it**:
This PR updates local and remote dev setups to:
- `etcd v3.4.14`
- `K8s v1.20.2`

**Release note**:

```feature developer
Nodeless local dev setups now run with `etcd v3.4.14` and `K8s v1.20.2`.
```

### CODE REVIEW

File: hack/local-development/local-garden/run-kube-apiserver [hunk 1]
```diff
@@ -13,7 +13,7 @@ LABEL=${1:-local-garden}
 
 SCRIPTPATH="$( cd "$(dirname "$0")" ; pwd -P )"
 
-IMAGE=k8s.gcr.io/kube-apiserver:v1.17.9
+IMAGE=k8s.gcr.io/kube-apiserver:v1.19.7
```
Comment (rfranzke) [inline hack/local-development/local-garden/run-kube-apiserver]: Can we go directly to 1.20.2?

Comment (timuthy) [reply hack/local-development/local-garden/run-kube-apiserver]: Thanks for the suggestion @rfranzke. I tested it successfully and updated the PR.

**Review by rfranzke — APPROVED** [review]
/lgtm

**Pull request merged**